src/app/(auth)/login/page.tsx

```diff
@@ -15,19 +15,23 @@ import { ProfileContext } from '@/utils/ProfileProvider';
 export default function Login() {
   const auth = useAuth();
   const profile = useContext(ProfileContext);
+  if (!auth || !profile) {
+    throw new Error('Page must have auth and profile context defined');
+  }
+
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [errorMessage, setErrorMessage] = useState('');
   const [emailError, setEmailError] = useState('');
   const [passwordError, setPasswordError] = useState('');
-  const [finishLogin, setFinishLogin] = useState(false);
+  const [isLoggingIn, setIsLoggingIn] = useState(false);
+
   const { push } = useRouter();
   const validEmail = (e: string) => e !== '' && isEmail(e);
 
   useEffect(() => {
-    if (!auth) throw new Error('Auth must be defined');
-    if (auth.userId && !finishLogin) push(CONFIG.settings);
-  }, [auth, profile, push]);
+    if (auth.userId && !isLoggingIn) push(CONFIG.settings);
+  }, [auth, profile, push, isLoggingIn]);
 
   const handleSignIn = async () => {
     if (!auth) {
```

Comment:
I think we can remove this conditional check since we've already checked that auth is defined earlier.